src/components/formly.form.ts

```diff
@@ -42,9 +41,36 @@ export class FormlyForm extends FormlyCommon implements OnInit  {
     if (!this.form) {
       this.form = this.formBuilder.group({});
     }
+
+    this.registerFormControls(this.fields);
   }
 
   changeModel(event: FormlyValueChangeEvent) {
     this.model[event.key] = event.value;
   }
+
+  private registerFormControls(fields) {
+    fields.map(field => {
+      if (field.key && field.type) {
+        let componenType: any = this.formlyConfig.getType(field.type).component;
+        if (Array.isArray(field.validation)) {
+          let validators = [];
+          field.validation.map((validate) => {
+            validators.push(this.formlyConfig.getValidator(validate).validation);
+          });
+          field.validation = Validators.compose(validators);
+        }
+
+        if (componenType.createControl) {
+          this.form.addControl(field.key, componenType.createControl(this.model[field.key] || "", field));
+        } else {
+          this.form.addControl(field.key, new FormControl({ value: this.model[field.key] || "", disabled: field.templateOptions.disabled }, field.validation));
+        }
+      }
+
+      if (field.fieldGroup) {
+        this.registerFormControls(field.fieldGroup);
+      }
+    });
+  }
```

Comment:
Will there be a problem when new field configuration is dynamically added later (i.e. after ngInit) ? Not sure if original formy supports this.